Automattic/wp-calypso
# Gutenberg: modify BrowserURL to match Calypso routes

This fixes #27298 where the browser url was incorrectly updated to match wp-admin routes vs Calypso ones. There is a separate issue to update the document title appropriately in https://github.com/Automattic/wp-calypso/issues/27300

### Testing Instructions
- Visit http://calypso.localhost:3000/gutenberg
- Select a site
- in console type `state.ui.route.path` verify that this matches the current path
- Type and let an autosave occur

The browser address is updated to:
http://calypso.localhost:3000/gutenberg/post/{siteSlug}/{postId}

Instead of http://calypso.localhost:3000/gutenberg/post/post.php?post={postId}&action=edit

After the url updates, verify that `state.ui.route.path` in console matches
